x-pack/plugins/cases/server/services/attachments/operations/get.ts

```diff
@@ -0,0 +1,250 @@
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { SavedObject } from '@kbn/core/server';
+import type * as estypes from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import {
+  CASE_COMMENT_SAVED_OBJECT,
+  CASE_SAVED_OBJECT,
+  MAX_DOCS_PER_PAGE,
+} from '../../../../common/constants';
+import { buildFilter, combineFilters } from '../../../client/utils';
+import type {
+  AttachmentTotals,
+  AttributesTypeAlerts,
+  CommentAttributes as AttachmentAttributes,
+  CommentAttributesWithoutRefs as AttachmentAttributesWithoutRefs,
+  CommentAttributes,
+} from '../../../../common/api';
+import { CommentType } from '../../../../common/api';
+import type {
+  AttachedToCaseArgs,
+  BulkOptionalAttributes,
+  GetAttachmentArgs,
+  ServiceContext,
+} from '../types';
+import {
+  injectAttachmentAttributesAndHandleErrors,
+  injectAttachmentSOAttributesFromRefs,
+} from '../../so_references';
+
+type GetAllAlertsAttachToCaseArgs = AttachedToCaseArgs;
+
+export class AttachmentGetter {
+  constructor(private readonly context: ServiceContext) {}
+
+  public async bulkGet(
+    attachmentIds: string[]
+  ): Promise<BulkOptionalAttributes<CommentAttributes>> {
+    try {
+      this.context.log.debug(
+        `Attempting to retrieve attachments with ids: ${attachmentIds.join()}`
+      );
+
+      const response =
+        await this.context.unsecuredSavedObjectsClient.bulkGet<AttachmentAttributesWithoutRefs>(
+          attachmentIds.map((id) => ({ id, type: CASE_COMMENT_SAVED_OBJECT }))
+        );
+
+      return {
+        saved_objects: response.saved_objects.map((so) =>
+          injectAttachmentAttributesAndHandleErrors(
+            so,
+            this.context.persistableStateAttachmentTypeRegistry
+          )
+        ),
+      };
+    } catch (error) {
+      this.context.log.error(
+        `Error retrieving attachments with ids ${attachmentIds.join()}: ${error}`
+      );
+      throw error;
+    }
+  }
+
+  public async getAttachmentIdsForCases({ caseIds }: { caseIds: string[] }) {
+    try {
+      this.context.log.debug(
+        `Attempting to retrieve attachments associated with cases: [${caseIds}]`
+      );
+
+      const finder = this.context.unsecuredSavedObjectsClient.createPointInTimeFinder({
+        type: CASE_COMMENT_SAVED_OBJECT,
+        hasReference: caseIds.map((id) => ({ id, type: CASE_SAVED_OBJECT })),
+        sortField: 'created_at',
+        sortOrder: 'asc',
+        /**
+         * We only care about the ids so to reduce the data returned we should limit the fields in the response. Core
+         * doesn't support retrieving no fields (id would always be returned anyway) so to limit it we'll only request
+         * the owner even though we don't need it.
+         */
+        fields: ['owner'],
```

Comment:
I know this was just moved around but I was curious why the `'owner'` field was here and read the comment 😅 

Won't `rootSearchFields: ['_id']` work here?